src/main/java/study1/step3/CarList.java

```diff
@@ -0,0 +1,24 @@
+package study1.step3;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CarList {
+    private final List<Car> carList = new ArrayList<>();
+    Decision moveDecisionMaker;
+
+    CarList(int carCnt, Decision moveDecisionMaker){
+        this.moveDecisionMaker = moveDecisionMaker;
+        for (int i = 0; i < carCnt; i ++){
+            carList.add(new Car());
+        }
+    }
```

Comment:
앞서 언급한 내용을 코드로 표현하면 다음과 같습니다.
```suggestion
    private CarList(Decision moveDecisionMaker, List<Car> carList){
        this.moveDecisionMaker = moveDecisionMaker;
        this.carList = carList;
    }
    
    public static CarList of(Decision moveDecisionMaker, List<Car> carList){
        return new CarList(moveDecisionMaker, carList);
    }

    public static CarList of(Decision moveDecisionMaker, int carCnt) {
      return new CarList(
        moveDecisionMaker,
        IntStream.range(0, carCnt)
                 .mapToObj(Car::new)
                 .collect(Collectors.toList())
      );
    }
```

Reply:
function 이름을 of로 하는 이유가 있나요??
static of로 쓰면 어떻게 호출해줘야하는건가요?

Reply:
@JunilHwang

Reply:
아 보다보니 이해했습니다. ㅋㅋㅋㅋ 
of(decision, 라운드 숫자) 선언하면 of(decision, list)를 선언하고 
of(decision이 list)가 생성자를 불러오는 구조네요

Reply:
of(decision, list) 함수를 빼고 of(decision, 라운드 숫자)에서 바로 생성자를 잡아주지 않는 이유는 객체를 확실히 나누기 위함인가요?

Reply:
@sunup1992 넵 말씀하신 내용이 맞습니다.